mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -1341,6 +1338,41 @@ class PreferencesActivity : AbstractBaseActivity() {
             }
         }
 
+        override fun onLeaveAndSave() {
+            Log.d(TAG, "Save tile $tileId")
+            val context = preferenceManager.context
+            val data: TileData? = if (enabledPref.isChecked) {
+                val itemName = itemAndStatePref.item
```

Comment:
Is there a reason for constructing the `TileData` instance slightly different than above here? I think it would be preferable to have a common method for constructing it.

Reply:
Yes, there's a reason. Here I only want to create the `TileData` only if no value is null or empty. Above I don't care if values are missing, I just need a `TileData` to compare it to the saved one.

Reply:
Yes, I understand that part, but can't you create the `TileData` instance and check validity of its members afterwards?

Reply:
Yes, that's probably better.